### PR TITLE
Pass the current AdminUser to all routes that require authentication

### PR DESCRIPTION
**Pull request by oshoham**
* Encode the current logged in `AdminUser`'s id in the auth token instead of their bot's name
* Add express middleware that finds the current `AdminUser` based on the id decoded from their auth token

### CODE REVIEW

File: backend/app/methods/adminMethods.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ const { logMessage } = require('../utilities/logHelper')
 exports.newAdmin = async function(req, res) {
 
   // only callparty user can create admin
-  const bot = req.user.bot
+  const bot = req.adminUser.bot
```
Comment (mhfowler) [inline backend/app/methods/adminMethods.js]: nice

File: backend/app/methods/adminMethods.js [hunk 2]
```diff
@@ -38,10 +38,10 @@ exports.newAdmin = async function(req, res) {
 }
 
 exports.getCurrentAdmin = async function(req, res) {
-  const bot = req.user.bot
-  const currentAdmin = await AdminUser.findOne({bot: bot})
+  const bot = req.adminUser.bot
+  const currentAdmin = await AdminUser.findOne({ bot: bot })
```
Comment (mhfowler) [inline backend/app/methods/adminMethods.js]: wait, can we just pass back req.adminUser instead of regetting?

File: backend/app/utilities/auth.js [hunk 3]
```diff
@@ -20,10 +20,18 @@ async function handleAuthTokenRequest(req, res) {
     return res.sendStatus(401)
   }
 
-  const token = jwt.sign({ bot: adminUser.bot }, process.env.JWT_SECRET, { expiresIn: '2h' })
+  const token = jwt.sign({ adminUserId: adminUser._id }, process.env.JWT_SECRET, { expiresIn: '2h' })
   res.json({ token: token })
 }
 
+async function adminUserMiddleware(req, res, next) {
+  if (req.user && req.user.adminUserId) {
```
Comment (mhfowler) [inline backend/app/utilities/auth.js]: nice

File: backend/app/methods/adminMethods.js (hunk 2 above)
Comment (oshoham) [inline backend/app/methods/adminMethods.js]: regretting?

Comment (oshoham) [inline backend/app/methods/adminMethods.js]: oh re-getting, misread that, yes we can do that

**Pull request merged**